File: sdks/node/examples/agent_groq.ts
import { Agent } from "alith";

const agent = new Agent({
  model: "llama3-70b-8192",
  apiKey: "Your API Key",
  baseUrl: "https://api.groq.com/openai/v1",
  preamble:
    "You are a calculator here to help the user perform arithmetic operations. Use the tools provided to answer the user question.",
});
(async () => {
  console.log(await agent.prompt("Calculate 10 - 3"));
})();

File: sdks/rust/examples/agent_groq.rs
use alith::{Agent, Chat, LLM};

#[tokio::main]
async fn main() {
    let model = LLM::openai_compatible_model(
        "Your-api-key",
        "https://api.groq.com/openai/v1",
        "llama-3.1-8b-instant",
    )
    .expect("Failed to create model");

    let agent = Agent::new("calculator agent", model)
        .preamble("You are a calculator here to help the user perform arithmetic operations. Use the tools provided to answer the user question.");

    let response = agent
        .prompt("Calculate 10 - 3")
        .await
        .expect("Failed to get response");

    println!("{}", response);
}
